cmd/snap-update-ns/bootstrap.go

```diff
@@ -69,47 +67,38 @@ func BootstrapError() error {
 
 // END IMPORTANT
 
-// readCmdline is a wrapper around the C function read_cmdline.
-func readCmdline(buf []byte) C.ssize_t {
-	return C.read_cmdline((*C.char)(unsafe.Pointer(&buf[0])), C.size_t(cap(buf)))
-}
-
-// findSnapName parses the argv-like array and finds the 1st argument.
-func findSnapName(buf []byte) *string {
-	if ptr := C.find_snap_name((*C.char)(unsafe.Pointer(&buf[0]))); ptr != nil {
-		str := C.GoString(ptr)
-		return &str
+func makeArgv(args []string) []*C.char {
+	argv := make([]*C.char, len(args)+1)
+	for i, arg := range args {
+		argv[i] = C.CString(arg)
 	}
-	return nil
+	return argv
 }
 
-// findFirstOption returns the first "-option" string in argv-like array.
-func findFirstOption(buf []byte) *string {
-	if ptr := C.find_1st_option((*C.char)(unsafe.Pointer(&buf[0]))); ptr != nil {
-		str := C.GoString(ptr)
-		return &str
+func freeArgv(argv []*C.char) {
+	for _, arg := range argv {
+		C.free(unsafe.Pointer(arg))
```

Comment:
Technically we don't have to free the last element, but free(NULL) is a no-op so ok.